modules/sdk/gradle-plugins-defaults/src/main/java/com/liferay/gradle/plugins/defaults/LiferayOSGiDefaultsPlugin.java

```diff
@@ -2953,6 +3040,8 @@ private void _configureTaskFindBugs(FindBugs findBugs) {
 	private void _configureTaskJar(Project project, boolean testProject) {
 		Jar jar = (Jar)GradleUtil.getTask(project, JavaPlugin.JAR_TASK_NAME);
 
+		jar.dependsOn(GENERATE_POM_INFO_TASK_NAME);
```

Comment:
Instead of the task name, ask for the task itself as a parameter.

Reply:
In the next part of this method it looks up a task by name. 

```if (testProject) {
			jar.dependsOn(JavaPlugin.TEST_CLASSES_TASK_NAME);```
Do we need to update that as well?